LoopKitUI/Views/ExpandableDatePicker.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  ExpandableDatePicker.swift
+//  LoopKitUI
+//
+//  Created by Anna Quinlan on 8/12/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import SwiftUI
+
+public struct ExpandableDatePicker: View {
+    @State var date: Date = Date()
+    let pickerRange: ClosedRange<Date>
+    let text: String
+    var onUpdate: (Date) -> Void
+    
+    public init (
+        with date: Date,
+        pickerRange: ClosedRange<Date>? = nil,
+        text: String = "",
+        onUpdate: @escaping (Date) -> Void
+    ) {
+        let today = Date()
+        self.pickerRange = pickerRange ?? today.addingTimeInterval(-.hours(6))...today.addingTimeInterval(.hours(6))
+        self.text = text
+        self.onUpdate = onUpdate
+        self.date = date
+    }
+    
+    public var body: some View {
+        ZStack(alignment: .topLeading) {
+            // ANNA TOOD: fix buggy animations
+            DatePicker(
+                "",
+                selection: $date.onChange(onUpdate),
```

Comment:
This is a little bit hacky, but since bindings can't be passed into a child view and then into a date picker, this is the best solution I was able to come up with